pkg/tfvars/internal/cache/cache.go

```diff
@@ -96,6 +96,21 @@ func cacheFile(reader io.Reader, filePath string, sha256Checksum string) (err er
 		}
 	}()
 
+	// Make sure that the temp file does not exist after completion of the function
+	defer func() {
```

Comment:
so if we have the lock on the file above `<file>.lock` and we have a tmp for that, should we just overwrite it or block it away?

or we could maybe create a tmp file, not anchored on the filename at all.

cc @wking because he did the initial implementation.

Reply:
I like the idea of giving temporary files random names!

Reply:
> ...should we just overwrite it or block it away?

I'm having trouble parsing this.  Can you elaborate?

> or we could maybe create a tmp file, not anchored on the filename at all.

Anchored on the filename is because we want to be in the same directory to setup for an atomic rename.

Reply:
> > ...should we just overwrite it or block it away?
> 
> I'm having trouble parsing this. Can you elaborate?

since we know that we own the lock on the end file, we can either overwrite the file.tmp or remove the file.tmp if it exists and create a new one.

> 
> > or we could maybe create a tmp file, not anchored on the filename at all.
> 
> Anchored on the filename is because we want to be in the same directory to setup for an atomic rename.

hhh, so rename is only atomic in same directory.. _need to read a bit on it_

Reply:
Oh, I think I understand "should we just overwrite it", and I think "yes, if we have the flock, we should be clobbering any pre-existing `.tmp` file".  That's somewhat orthogonal to this PR's current attempt to remove any dangling `.tmp` file on exit.  We could do both, although clobbering pre-existing `.tmp` files is the only thing that will help in the "earlier installer was killed without having time to clean up" case.

Reply:
technically the goal of the PR should be to fix https://github.com/openshift/installer/issues/1668

and I think in that case, we should probably just go with, overwrite the tmp file instead of trying to clean it up.. 
@Fedosin wdyt?

Reply:
@abhinavdahiya funny you said that, because one of the requirements for the checksum validation work is "Corrupted data should remain on disk for further investigation of the cause of failure", so I think you're right here.